services/controlplane/client_test.go

```diff
@@ -342,3 +343,61 @@ func TestRetriesTimeout(t *testing.T) {
 		})
 	}
 }
+
+func TestGetDestinationSSHKeyPair(t *testing.T) {
+	var count int
+	mockServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if count == 0 { // failing the first call to test retries
```

Comment:
@saurav-malani I added this to test the retries behaviour. The first call always fails.